src/store/audienceCache.ts

```diff
@@ -0,0 +1,58 @@
+import { StorageKeys, AudienceDefinition, CachedAudienceMetaData } from 'types';
+import { storage, timeStampInSecs } from '../utils';
+
+class AudienceCache {
+  cachedAudiences: AudienceDefinition[];
+  cachedAudiencesMetaData: CachedAudienceMetaData;
+
+  constructor() {
+    this.cachedAudiences = [];
+    this.cachedAudiencesMetaData = {
+      timeChecked: 0,
+      audiences: [],
+    };
+    this._load();
+  }
+
+  _load() {
+    const audiences: AudienceDefinition[] =
+      storage.get(StorageKeys.CACHED_AUDIENCES) || [];
+    const audiencesMetaData: CachedAudienceMetaData =
+      storage.get(StorageKeys.CACHED_AUDIENCE_META_DATA) || [];
+    this.cachedAudiences = audiences;
+    this.cachedAudiencesMetaData = audiencesMetaData;
```

Comment:
This overwrites the object you defined in the constructor, if nothing is in the store with `[]`